lib/shared/widgets/led_grid_widget.dart

```diff
@@ -0,0 +1,82 @@
+import 'package:flutter/material.dart';
+import 'package:soccer_app_flutter/shared/models/practice_menu.dart';
+import 'package:soccer_app_flutter/shared/utils/led_utils.dart';
+
+/// LEDグリッドウィジェット
+/// LEDの表示ロジックを共通化したウィジェット
+class LedGridWidget extends StatelessWidget {
+  final PracticeMenu menu;
+  final int currentPhaseIndex;
+  final LedDisplayConfig config;
+
+  const LedGridWidget({
+    super.key,
+    required this.menu,
+    required this.currentPhaseIndex,
+    required this.config,
+  });
+
+  @override
+  Widget build(BuildContext context) {
+    return Wrap(
+      spacing: config.spacing,
+      runSpacing: config.spacing,
+      alignment: WrapAlignment.center,
+      children: List.generate(menu.ledCount, (ledIndex) => _buildLed(ledIndex)),
+    );
+  }
+
+  /// 個別のLEDを構築
+  Widget _buildLed(int ledIndex) {
+    final String colorName = _getColorName(ledIndex);
+    final Color ledColor = LedUtils.colorNameToColor(colorName);
```

Comment:
作成したデータクラスを使うと，こんな感じで書き換えられると思います
```suggestion
    final Color ledColor = LedUtils.colorNameToColor(LedColor.fromLabel(colorName));
```